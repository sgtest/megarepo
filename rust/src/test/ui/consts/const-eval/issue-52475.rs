// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![feature(const_let)]

fn main() {
    let _ = [(); {
        //~^ WARNING Constant evaluating a complex constant, this might take some time
        let mut x = &0;
        let mut n = 0;
        while n < 5 { //~ ERROR constant contains unimplemented expression type
            n = (n + 1) % 5; //~ ERROR evaluation of constant value failed
            x = &0; // Materialize a new AllocId
        }
        0
    }];
}
