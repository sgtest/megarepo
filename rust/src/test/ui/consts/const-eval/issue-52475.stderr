error[E0658]: `while` is not allowed in a `const`
  --> $DIR/issue-52475.rs:5:9
   |
LL | /         while n < 5 {
LL | |
LL | |             n = (n + 1) % 5;
LL | |             x = &0; // Materialize a new AllocId
LL | |         }
   | |_________^
   |
   = note: see issue #52000 <https://github.com/rust-lang/rust/issues/52000> for more information
   = help: add `#![feature(const_loop)]` to the crate attributes to enable
   = help: add `#![feature(const_if_match)]` to the crate attributes to enable

error[E0080]: evaluation of constant value failed
  --> $DIR/issue-52475.rs:7:17
   |
LL |             n = (n + 1) % 5;
   |                 ^^^^^^^^^^^ exceeded interpreter step limit (see `#[const_eval_limit]`)

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0080, E0658.
For more information about an error, try `rustc --explain E0080`.
